plugin/src/pages/AIPage/index.tsx

```
@@ -0,0 +1,210 @@
+  useEffect(() => {
+    /**
+     * Fired from main.ts after nodes are classified, grouped and (optionally)
+     * images exported. We now call the server.
+     */
+    const unsubBatchReady = on<AIBatchReadyHandler>('AI_BATCH_READY', async (groups: AIRenameGroup[]) => {
+      // Count total target nodes across all groups
+      const total = groups.reduce((acc, g) => acc + g.targetNodes.length, 0);
+      setProgress({ done: 0, total });
+
+      // Mark all as pending
+      groups.forEach((g) => g.targetNodes.forEach((t) => setRenameStatus(t.nodeId, 'pending')));
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)